src/inflate.js

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable no-underscore-dangle */
+
+const zlib = require('zlib');
+const stream = require('stream');
+
+/**
+ * Creates an intelligent inflate stream, that can handle deflate responses from older servers,
+ * which do not send the correct GZip headers in the response. See http://stackoverflow.com/a/37528114
+ * for details on why this is needed.
+ *
+ * @param {Object=} options - Are passed to the underlying `Inflate` or `InflateRaw` constructor.
+ *
+ * @returns {*}
+ */
+module.exports.createInflate = function (options) {
+    return stream.Transform({
+
+        /**
+         * Override the _transform method of the stream.
+         *
+         * @param chunk
+         * @param encoding
+         * @param callback
+         */
+        transform(chunk, encoding, callback) {
+            const self = this;
+            if (!self._inflate) {
+                // If the response stream does not have a valid deflate header, use `InflateRaw`
+                if ((Buffer.from(chunk, encoding)[0] & 0x0F) === 0x08) { //eslint-disable-line
```

Comment:
To reduce performance overheads, maybe you can slice `chunk` to be just a few characters here? That should be enough for the test

Reply:
I also think that the chunk should be always buffer, so using Buffer.from is also unnecessary for our implementation. -  from node js docu - chunk <Buffer> | <string> | <any> The Buffer to be written, converted from the string passed to stream.write(). If the stream's decodeStrings option is false or the stream is operating in object mode, the chunk will not be converted & will be whatever was passed to stream.write().